src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesignerIntermediateClass.cs

```diff
@@ -40,25 +40,28 @@ public partial class Resource : %BASECLASS% {
 //------------------------------------------------------------------------------
 namespace %NAMESPACE%
 
-type Resource = %BASECLASS%
+type %MODIFIER% Resource = %BASECLASS%
 ";
 
 		public string Namespace { get; set; }
+		public string Modifier { get; set; } = "public";
 		public bool IsApplication { get; set; } = false;
 		public ITaskItem OutputFile { get; set; }
 		public override bool RunTask ()
 		{
-			string ns = IsApplication ? ResourceDesignerConstants : ResourceDesigner;
+			string baseClass = IsApplication ? ResourceDesignerConstants : ResourceDesigner;
+			string modifier = IsApplication ? Modifier : "public";
```

Comment:
Why can't library projects override visibility?

Reply:
Because the `Resource` class in the Designer.cs file has historically been public . Users have used it from other libraries in the past. So people have done `Library2.Resource.Layout.foo` rather than use the one in `Library1.Resource.Layout.foo`. This is something we have seen. 

I'm ok with removing this restriction and allowing the user to decide (but default to public).

Reply:
On the one hand, *yes*, historically they've been public.

But also we have the [`$(AndroidGenerateResourceDesigner)`](https://learn.microsoft.com/en-us/dotnet/android/building-apps/build-properties#androidgenerateresourcedesigner) MSBuild property alongside 953cd1939d309e0affe610728a79ede865a3bb75, so we already have a way to control *generation* of the `Resource.designer.cs` file.  Allowing control of the *visibility* of the `Resource` type feels like a straightforward expansion.